src/base/bittorrent/session.cpp

```diff
@@ -2377,10 +2377,10 @@ bool Session::loadTorrent(LoadTorrentParams params)
 #else
     const auto id = TorrentID::fromInfoHash(hasMetadata ? p.ti->info_hash() : p.info_hash);
 #endif
-    m_loadingTorrents.insert(id, params);
 
-    // Adding torrent to BitTorrent session
+    // Adding torrent to libtorrent session
     m_nativeSession->async_add_torrent(p);
+    m_loadingTorrents.emplace(id, std::move(params));
```

Comment:
Seems the old statement order is safer, at least without inspecting deeper.

Reply:
>Seems the old statement order is safer

A groundless statement... Or do you have some special opinion about their safety?

Reply:
After a closer look, now I understand why you switched the order.

>Or do you have some special opinion about their safety?

Maybe it doesn't apply here, but the general idea is to do whatever action that is locally to us first and kick off foreign (async) actions later, so that our local state would be less affected when foreign actions runs amok unexpectedly.

Reply:
> the general idea is to do whatever action that is locally to us first and kick off foreign (async) actions later, so that our local state would be less affected when foreign actions runs amok unexpectedly.

In general, yes. But here we are exactly aware of the actions order. Although this is not obvious to an "outsider".
I can change it in the following way:
```c++
const auto iter = m_loadingTorrents.emplace(id, std::move(params));
m_nativeSession->async_add_torrent(iter->ltAddTorrentParams);
```

Reply:
>I can change it in the following way

Done.